### PR TITLE
deep decode proposal actions

### PR DESCRIPTION
**Pull request by dan13ram**
## GitHub Issue

N/A
Discussion: https://discord.com/channels/709210493549674598/1082764707141795960/1154518182837637150

## Changes

- Adds a method to "deep" decode proposal actions in `libs/tx-builder/src/utils/deepDecoding.ts`

- Replaces the existing `decodeProposalActions` method with the new `deepDecodeProposalActions` method in `libs/moloch-v3-macro-ui/src/components/ProposalDetails/ProposalDetailsContainer.tsx`

- Renders the deeper level decoded actions in `libs/moloch-v3-macro-ui/src/components/ProposalActionData/ProposalActionData.tsx` recursively. GIF for the same attached below: 

decoded-actions

## Packages Added

- "@shazow/whatsabi": "^0.8.5"
- "ethers-multisend": "^2.4.0"

## Checks

Before making your PR, please check the following:

- [x] Critical lint errors are resolved
- [x] App runs locally
- [x] App builds locally (run the build command for _any impacted package_ and check for any errors before the PR)

Note: git push is running all these checks by default so I've not run them myself except for testing the app locally which of course runs

### CODE REVIEW

File: libs/tx-builder/src/utils/deepDecoding.ts [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+      to: to,
+      operation: OperationType.DelegateCall,
+      name: result.functionName,
+      value: decodeValue(value),
+      params: result.inputs,
+      decodedActions,
+    };
+  }
+
+  if (result.functionName.toLowerCase().includes('exec')) {
```
Comment (dan13ram) [inline libs/tx-builder/src/utils/deepDecoding.ts:228]: I'm not very happy with how this is done, so please critique this and suggest better ways to solve this.

One clean way I can think of is to support only known functions such as `execTransactionFromModule` and slowly add support for similar functions in the wild in future patches.

Comment (skuhlmann) [reply libs/tx-builder/src/utils/deepDecoding.ts:228]: Yeah that is tough.

I like your idea of supporting the known functions. Maybe we could make a constant with the known function names and do the if statement on that? Would be easy then to add more.

```js
const knownFunctionsNeedsBetterName = ["execTransactionFromModule", "executeAsBaal"];

...

if (knownFunctionsNeedsBetterName.some(functionName => functionName === result.functionName)) {};
```

What do you think?

File: libs/tx-builder/src/utils/deepDecoding.ts [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+      to: to,
+      operation: operation || OperationType.Call,
+      name: `${HAUS_NETWORK_DATA[chainId]?.symbol} Transfer`,
+      value: decodeValue(value),
+      params: [],
+      decodedActions: [],
+    };
+  }
+
+  const { abi } = await whatsabi.autoload(to, {
```
Comment (skuhlmann) [inline libs/tx-builder/src/utils/deepDecoding.ts:193]: I've never used whatsabi. looks powerful.

After this we might want to explore replacing all our abi fetching with this.

**Review by skuhlmann — CHANGES_REQUESTED** [review]
This is looking good to me!

I think we ought to just make the change around how we identify when we need to the decode deeper - that function name list and check like you suggested - and then we'll be good to go.

File: libs/tx-builder/src/utils/deepDecoding.ts (hunk 2 above)
Comment (dan13ram) [reply libs/tx-builder/src/utils/deepDecoding.ts:193]: Thanks to @dekanbro for the library!

File: libs/tx-builder/src/utils/deepDecoding.ts (hunk 1 above)
Comment (dan13ram) [reply libs/tx-builder/src/utils/deepDecoding.ts]: @skuhlmann I've pushed a change to handle this more cleanly. Let me know what you think!

**Review by skuhlmann — APPROVED** [review]

**Pull request merged**